location-service/src/main/java/com/movie/recommender/location/service/LocationService.java

```diff
@@ -125,17 +126,22 @@ public List<CountryWithCitiesDTO> getAllCountriesAndCities() {
         log.info("Fetching all countries and cities");
 
         List<CountryCityDTO> countriesAndCities = countryRepository.getAllCountriesAndCities();
-        Map<String, List<String>> countriesCitiesMap = new HashMap<>();
+
+        Map<String, Map.Entry<String, List<String>>> countriesMap = new HashMap<>();
```

Comment:
whole logic can be simplified by using a single loop
instead of using a nested map structure `Map<String, Map.Entry<String, List<String>>>`, you could directly map the country name to a `CountryWithCitiesDTO`